core/storage/car/car.cpp

```diff
@@ -95,4 +114,72 @@ namespace fc::storage::car {
     }
     return makeCar(store, roots, cid_order);
   }
+
+  void writeUvarint(std::ostream &output, uint64_t value) {
+    output << common::span::bytestr(libp2p::multi::UVarint{value}.toBytes());
+  }
+
+  void writeHeader(std::ostream &output, const std::vector<CID> &roots) {
+    OUTCOME_EXCEPT(bytes, codec::cbor::encode(CarHeader{roots, CarHeader::V1}));
+    writeUvarint(output, bytes.size());
+    std::copy(bytes.begin(),
```

Comment:
Why not `output << common::span::bytestr(bytes)`?

Reply:
Fixed